lib/db_connection/connection_pool.ex

```diff
@@ -33,10 +34,38 @@ defmodule DBConnection.ConnectionPool do
     GenServer.call(pool, {:disconnect_all, interval}, :infinity)
   end
 
+  @doc """
+  Returns connection module used by the given connection pool process.
+
+  If the given process is not a connection pool, `:error` is returned.
+  """
+  @spec connection_module(pid() | atom()) :: {:ok, module()} | :error
+  def connection_module(pool) when is_pid(pool) or is_atom(pool) do
+    with {:ok, pid} <- to_pid(pool),
+         {:dictionary, dictionary} <- Process.info(pid, :dictionary),
+         {:ok, module} <- fetch_from_dictionary(dictionary, @connection_module_key),
+         do: {:ok, module},
+         else: (_ -> :error)
+  end
+
+  defp to_pid(nil), do: :error
+  defp to_pid(pid) when is_pid(pid), do: {:ok, pid}
+  defp to_pid(name) when is_atom(name), do: name |> Process.whereis() |> to_pid()
+  defp to_pid(_other), do: :error
+
+  defp fetch_from_dictionary(dictionary, key) do
+    Enum.find_value(dictionary, :error, fn
```

Comment:
You can use the keyword module? Otherwise, consider using `List.keyfind`!

Reply:
pdict is not a keyword list, as the keys may be `{:any, :thing}`. The advantage over `List.keyfind` is that if there's a `nil` value we also want it (in this case it doesn't matter though) :)